pkg/cvo/internal/operatorstatus.go

```diff
@@ -96,6 +96,10 @@ func (b *clusterOperatorBuilder) WithModifier(f resourcebuilder.MetaV1ObjectModi
 
 func (b *clusterOperatorBuilder) Do(ctx context.Context) error {
 	os := readClusterOperatorV1OrDie(b.raw)
+
+	if _, ok := payload.ClusterOperatorUpdateStartTimes[os.Name]; !ok {
+		payload.ClusterOperatorUpdateStartTimes[os.Name] = time.Now()
```

Comment:
Both [e2e][1] and [e2e-operator][2] died on:

```
level=fatal msg=Bootstrap failed to complete: failed to wait for bootstrapping to complete: timed out waiting for the condition
```

In the log bundle:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/486/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1352370358094663680/artifacts/e2e-agnostic/ipi-install-install/log-bundle-20210121224256.tar | tar -xOz log-bundle-20210121224256/bootstrap/containers/cluster-version-operator-05aa025888b3bee7a1b52ba61bef85baace11a35eae7a3f9e6fe0f0174fbbaaf.log
...
E0121 22:42:32.828219       1 runtime.go:78] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 216 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1856060, 0x1c47440)
        /go/src/github.com/openshift/cluster-version-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/openshift/cluster-version-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x1856060, 0x1c47440)
        /usr/lib/golang/src/runtime/panic.go:969 +0x1b9
github.com/openshift/cluster-version-operator/pkg/cvo/internal.(*clusterOperatorBuilder).Do(0xc001758000, 0x1ca5440, 0xc000d4ecc0, 0xc001758000, 0xc001758000)
        /go/src/github.com/openshift/cluster-version-operator/pkg/cvo/internal/operatorstatus.go:101 +0x489
github.com/openshift/cluster-version-operator/pkg/cvo.(*resourceBuilder).Apply(0xc0019063f0, 0x1ca5440, 0xc000d4ecc0, 0xc001a82be0, 0x3, 0x0, 0x0)
        /go/src/github.com/openshift/cluster-version-operator/pkg/cvo/cvo.go:713 +0xcb
github.com/openshift/cluster-version-operator/pkg/cvo.(*SyncWorker).apply.func1(0x1ca5440, 0xc000d4ecc0, 0xc000ce6368, 0x9, 0x393, 0x0, 0x1a6c757)
        /go/src/github.com/openshift/cluster-version-operator/pkg/cvo/sync_worker.go:747 +0x299
github.com/openshift/cluster-version-operator/pkg/payload.RunGraph.func3(0xc0009ff280, 0xc001e8da40, 0xc001629de0, 0xc001e8daa0, 0x1ca5440, 0xc000d4ecc0, 0x6)
        /go/src/github.com/openshift/cluster-version-operator/pkg/payload/task_graph.go:487 +0xed
created by github.com/openshift/cluster-version-operator/pkg/payload.RunGraph
        /go/src/github.com/openshift/cluster-version-operator/pkg/payload/task_graph.go:477 +0x32f
I0121 22:42:32.828247       1 task_graph.go:486] Running 23 on worker 5
panic: assignment to entry in nil map [recovered]
        panic: assignment to entry in nil map
        ...
```

You have the `make(map...)` call in `sync_worker.go`, but maybe add a fallback here to guard against that `apply()` pathway?  Might be getting called via a parallel thread; I haven't traced it out yet.  Probably worth tracing out in case we need to guard it with a lock...  Actually, we definitely want to guard it with a lock, see [here][3] and [here][4].

There may also be alternatives to a map that are more convenient to concurrency.  Maybe we could inspect the manifest around [here][5] and feed "I'm about to attempt $CLUSTER_VERSION_OPERATOR_NAME" into a local channel that we collect into the map in a parallel goroutine (so only that one goroutine would be writing to the map)?  I dunno.  Maybe we ditch the "40m" thing entirely as "not worth the trouble" ;).

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/486/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1352370358094663680
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/486/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator/1352370358128218112
[3]: https://blog.golang.org/maps#TOC_6.
[4]: https://golang.org/doc/faq#atomic_maps
[5]: https://github.com/openshift/cluster-version-operator/blob/f9c42b254b2103be17ca26519f500490a80db125/pkg/cvo/sync_worker.go#L768

Reply:
added lock